doc/tutorials/sgx_ virtualization.rst

```diff
@@ -0,0 +1,213 @@
+-------------------------
+
+This is read-only since SGX LC is currently not supported.
+
+SGXOWNEREPOCH[0-1]
+------------------
+
+* This is a 128-bit external entropy value for key derivation of an enclave.
+* These MSRs are at the package level; they cannot be controlled by the VM.
+
+EPC Virtualization
+==================
+
+* EPC resource is statically partitioned according to the configuration of the EPC size of VMs.
+* During platform initialization, the physical EPC section information is collected via CPUID. SGX initialization function allocates EPC resource to VMs according to the EPC config in VM configurations.
+* If enough EPC resource is allocated for the VM, assign the GPA of the EPC section.
+* EPC resource is allocated to the Non-SOS VM; the EPC base GPA is specified by the EPC config in the VM configuration.
+* The corresponding range of memory space should be marked as reserved in E820.
+* During initialization, the mapping relationship of EPC HPA and GPA is saved for building the EPT table later when the VM is created.
+
+Enclave System Function Handling
+********************************
+
+A new "Enable ENCLS exiting" control bit (bit 15) is defined in the secondary processor-based VM execution control.
+
+* 1-Setting of "Enable ENCLS exiting" enables ENCLS-exiting bitmap control, which is a new 64-bit ENCLS-exiting bitmap control field added to VMX VMCS (0202EH) to control VMEXIT on ENCLS leaf functions.
+* ACRN does not emulate ENCLS leaf functions and will not enable ENCLS exiting.
+
+ENCLS[ECREATE]
+==============
+
+* The enclave execution environment is heavily influenced by the value of ATTRIBUTES in the enclave's SECS.
+* When ECREATE is executed, the processor will check and verify that the enclave requirements are supported on the platform. If not, ECREATE will generate a #GP.
+* The hypervisor can present the same extended features to Guest as the hardware. However, if the hypervisor hides some extended features that the hardware supports from the VM/guest, then if the hypervisor does not trap ENCLS[ECREATE], ECREATE may succeed even if the ATTRIBUTES the enclave requested is not supported in the VM.
+* Fortunately, ENCLU[EENTER] will fault if SECS.ATTRIBUTES.XFRM is not a subset of XCR0 when CR4.OSXSAVE = 1.
+* XCR0 is controlled by the hypervisor in ACRN; if the hypervisor hides some extended feature from the VM/guest, then ENCLU[EENTER] will fault if the enclave requests a feature that the VM does not support if the hypervisor does not trap/emulate ENCLS[ECREATE].
+* Above all, the security feature is not compromised if the hypervisor does not trap ENCLS[ECREATE] to check the attributes of the enclave.
+
+Other VMExit Control
+********************
+
+RDRAND exiting
+==============
+
+* ACRN allows Guest to use RDRAND/RDSEED instruction but does not set "RDRAND exiting" to 1.
+
+PAUSE exiting
+=============
+
+* ACRN does not set "PAUSE exiting" to 1.
+
+Future Development
+******************
+
```

Comment:
Need some words here to say we're not promising these features any time soon, right?

Reply:
no timeframe yet.